edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentCommand.cs

```diff
@@ -98,7 +121,7 @@ async Task UpdateImagePullSecrets(Dictionary<string, ImagePullSecret> imagePullS
 
                             return await this.client.ReplaceNamespacedSecretAsync(
                                 newSecret,
-                                imagePullSecret.Key,
+                                imagePullSecret.Name,
```

Comment:
Why change to Name instead?

Reply:
It used to be a KeyValuePair here but I created ImagePullSecret and it transformed to collections instead of Dictionary. ImagePullSecret class encapsulates some logic for name genaration inside. So Name is more meaningful name for the field.

Reply:
I see.